packages/vite/src/node/utils.ts

```diff
@@ -963,21 +964,23 @@ export function getHash(text: Buffer | string): string {
   return createHash('sha256').update(text).digest('hex').substring(0, 8)
 }
 
+const _dirname = path.dirname(fileURLToPath(import.meta.url))
+
 export const requireResolveFromRootWithFallback = (
   root: string,
   id: string,
 ): string => {
-  const paths = _require.resolve.paths?.(id) || []
-  // Search in the root directory first, and fallback to the default require paths.
-  paths.unshift(root)
-
-  // Use `resolve` package to check existence first, so if the package is not found,
+  // check existence first, so if the package is not found,
   // it won't be cached by nodejs, since there isn't a way to invalidate them:
   // https://github.com/nodejs/node/issues/44663
-  resolve.sync(id, { basedir: root, paths })
+  const found = resolvePackageData(id, root) || resolvePackageData(id, _dirname)
+  if (!found) {
+    throw new Error(`${JSON.stringify(id)} not found.`)
```

Comment:
I think we also need to attach `error.code = 'MODULE_NOT_FOUND'` here as the try..catch blocks that uses this function provides better error message for that code specifically:

https://github.com/vitejs/vite/blob/fedb0803308f8d62ff88f300187f0379a4d49f91/packages/vite/src/node/plugins/css.ts#L1545-L1549

https://github.com/vitejs/vite/blob/fedb0803308f8d62ff88f300187f0379a4d49f91/packages/vite/src/node/plugins/terser.ts#L13-L17

There's also the opportunity to not throw errors here, but given this happens rarely, I think throwing errors still would be easier to handle.

Reply:
Good catch! I added `error.code`.